File: sdks/php/src/Types/HealthStatus.php
<?php

declare(strict_types=1);

namespace Blok\Blok\Types;

/**
 * HealthStatus represents the health status of the runtime.
 */
final class HealthStatus
{
    public function __construct(
        public string $status = 'healthy',
        public string $version = '1.0.0',
        /** @var string[] */
        public array $nodesLoaded = [],
    ) {}

    /**
     * Create a HealthStatus from an associative array.
     */
    public static function fromArray(array $data): self
    {
        return new self(
            status: $data['status'] ?? 'healthy',
            version: $data['version'] ?? '1.0.0',
            nodesLoaded: $data['nodes_loaded'] ?? [],
        );
    }

    /**
     * Convert to an associative array for JSON serialization.
     */
    public function toArray(): array
    {
        return [
            'status' => $this->status,
            'version' => $this->version,
            'nodes_loaded' => $this->nodesLoaded,
        ];
    }
}

File: sdks/csharp/src/Blok.Core/Types/HealthStatus.cs
using System.Text.Json.Serialization;

namespace Blok.Core.Types;

/// <summary>
/// HealthStatus represents the health status of the runtime.
/// </summary>
public class HealthStatus
{
    [JsonPropertyName("status")]
    public string Status { get; set; } = "healthy";

    [JsonPropertyName("version")]
    public string Version { get; set; } = "1.0.0";

    [JsonPropertyName("nodes_loaded")]
    public List<string> NodesLoaded { get; set; } = new();
}
